test/integration/src/main/java/io/pravega/test/integration/utils/TestUtils.java

```diff
@@ -0,0 +1,214 @@
+     * @param scope the scope
+     * @param stream the stream
+     * @param readerClientConfig the {@link ClientConfig} object to use to connect to the server
+     * @param readerGroup the name of the reader group application
+     * @return the event message
+     */
+    public static String readAMessageFromStream(String scope, String stream, ClientConfig readerClientConfig,
+                                                 String readerGroup) {
+        List<String> messages = readDataFromStream(scope, stream, 1, readerClientConfig, readerGroup);
+        if (messages == null || messages.size() == 0) {
+            return null;
+        } else {
+            return messages.get(0);
+        }
+    }
+
+    /**
+     * Returns the specified number of messages from the specified {@code scope}/{@code stream}.
+     *
+     * @param scope the scope
+     * @param stream the stream
+     * @param numMessages the number of event messages to write
+     * @param readerClientConfig the {@link ClientConfig} object to use to connect to the server
+     * @param readerGroup the name of the reader group application
+     * @return the event messages
+     * @throws NullPointerException if {@code scope} or {@code stream} or {@writerClientConfig} is null
+     * @throws IllegalArgumentException if {@code numMessages} < 1
+     * @throws RuntimeException if any exception is thrown by the client
+     */
+    public static List<String> readDataFromStream(@NonNull String scope, @NonNull String stream, int numMessages,
+                                            @NonNull ClientConfig readerClientConfig, @NonNull String readerGroup) {
+        Preconditions.checkArgument(numMessages > 0);
+
+        @Cleanup
+        EventStreamClientFactory readerClientFactory = EventStreamClientFactory.withScope(scope, readerClientConfig);
+        log.debug("Created the readerClientFactory");
+
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                .stream(Stream.of(scope, stream))
+                .disableAutomaticCheckpoints()
+                .build();
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, readerClientConfig);
+        readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
+        log.debug("Created reader group with name {}", readerGroup);
+
+        @Cleanup
+        EventStreamReader<String> reader = readerClientFactory.createReader(
+                "readerId", readerGroup,
+                new JavaSerializer<String>(), ReaderConfig.builder().initialAllocationDelay(0).build());
+        log.debug("Created an event reader");
+
+        // Keeping the read timeout large so that there is ample time for reading the event even in
+        // case of abnormal delays in test environments.
+        List<String> result = new ArrayList<>();
+        for (int i = 0; i < numMessages; i++) {
+            result.add(reader.readNextEvent(20000).getEvent());
```

Comment:
in case of a checkpoint event, you could get a null event. 
i think you should add event only when it is not a checkpoint event. 
or disable checkpoint in the readergroup config. 
otherwise that can lead to some tests to fail.

Reply:
We are assuming that the checkpoint is disabled in the reader group: 

https://github.com/pravega/pravega/blob/4788434e05b3cafc4dfc7e2bb2c56112ab6ac2ef/test/integration/src/main/java/io/pravega/test/integration/utils/TestUtils.java#L161-L164